### PR TITLE
support MySQL SET session variable (#24124)

### LINKED ISSUE #24124: Proxy can not support MySQL `SET @order_id := 1, @item_id := 2` statement
### Which version of ShardingSphere did you use?

[d0e096b](https://github.com/apache/shardingsphere/commit/d0e096ba72ac7af072131168d4b3602b4d6cd437)

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?

ShardingSphere-Proxy

### Expected behavior

```sql
mysql> SET @order_id := 1, @item_id := 2;
Query OK, 0 rows affected (0.12 sec)

mysql> SELECT @order_id, @item_id;
+-----------+----------+
| @order_id | @item_id |
+-----------+----------+
| 1      | 2     |
+-----------+----------+
1 row in set (0.06 sec)
```

### Actual behavior

```sql
mysql> SET @order_id := 1, @item_id := 2;
Query OK, 0 rows affected (0.12 sec)

mysql> SELECT @order_id, @item_id;
+-----------+----------+
| @order_id | @item_id |
+-----------+----------+
| NULL      | NULL     |
+-----------+----------+
1 row in set (0.06 sec)
```

### PR DESCRIPTION
**Pull request by iamhucong**
Fixes #24124.

Changes proposed in this pull request:
  -

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Comment by strongduanmu** [comment]
Hi @TeslaCN, can you help review this pr?

File: proxy/backend/core/src/main/java/org/apache/shardingsphere/proxy/backend/handler/admin/executor/DefaultSessionVariableHandler.java [hunk 1]
```diff
@@ -37,13 +37,18 @@ public abstract class DefaultSessionVariableHandler implements SessionVariableHa
...
-                .contains(variableName)) {
+                .contains(variableName)
+                || isNeedHandel(variableName)) {
             connectionSession.getRequiredSessionVariableRecorder().setVariable(variableName, assignValue);
         } else {
             log.debug("Set statement {} = {} was discarded.", variableName, assignValue);
         }
     }
     
+    protected boolean isNeedHandel(final String variableName) {
```
Comment (TeslaCN) [inline proxy/backend/core/src/main/java/org/apache/shardingsphere/proxy/backend/handler/admin/executor/DefaultSessionVariableHandler.java:48]: `Handel` or `Handle`?

File: proxy/backend/core/src/main/java/org/apache/shardingsphere/proxy/backend/handler/admin/executor/DefaultSessionVariableHandler.java [hunk 2]
```diff
@@ -37,13 +37,18 @@ public abstract class DefaultSessionVariableHandler implements SessionVariableHa
     @Override
     public final void handle(final ConnectionSession connectionSession, final String variableName, final String assignValue) {
         if (TypedSPILoader.findService(ReplayedSessionVariablesProvider.class, databaseType).map(ReplayedSessionVariablesProvider::getVariables).orElseGet(Collections::emptySet)
-                .contains(variableName)) {
+                .contains(variableName)
+                || isNeedHandel(variableName)) {
```
Comment (TeslaCN) [inline proxy/backend/core/src/main/java/org/apache/shardingsphere/proxy/backend/handler/admin/executor/DefaultSessionVariableHandler.java:41]: Cound we join these lines?

Comment (iamhucong) [reply proxy/backend/core/src/main/java/org/apache/shardingsphere/proxy/backend/handler/admin/executor/DefaultSessionVariableHandler.java]: Ok.

File: proxy/backend/core/src/main/java/org/apache/shardingsphere/proxy/backend/handler/admin/executor/DefaultSessionVariableHandler.java (hunk 1 above)
Comment (iamhucong) [reply proxy/backend/core/src/main/java/org/apache/shardingsphere/proxy/backend/handler/admin/executor/DefaultSessionVariableHandler.java]: have been rectified.

**Review by TeslaCN — APPROVED** [review]

**Pull request merged**